### PR TITLE
Resolve the full response

### PR DESCRIPTION
**Pull request by jderusse**
Provide a way to resolve the entire body, instead of just the first bytes.

That provide full async until the end of the response.

Partially fixes #376

### CODE REVIEW

File: src/Core/src/Response.php [hunk 1]
```diff
@@ -80,7 +88,17 @@ public function resolve(?float $timeout = null): bool
...
                 }
-                if ($chunk->isFirst()) {
+                if ($fullResponse) {
+                    if ($chunk->isLast()) {
+                        $this->streamable = false;
+
+                        break;
+                    }
+
+                    if ($this->streamable && '' !== $chunk->getContent()) {
```
Comment (Nyholm) [inline src/Core/src/Response.php]: Can you read a chunk multiple times?

Comment (jderusse) [reply src/Core/src/Response.php]: yes, you can read multiple time the same chunk, but can't rewind the stream

File: src/Core/src/Response.php [hunk 2]
```diff
@@ -53,15 +60,16 @@ public function __destruct()
...
      *
      * @return bool whether the request is executed or not
      *
      * @throws NetworkException
      * @throws HttpException
      */
-    public function resolve(?float $timeout = null): bool
+    public function resolve(?float $timeout = null, bool $fullResponse = false): bool
     {
         if (null !== $this->resolveResult) {
```
Comment (jderusse) [inline src/Core/src/Response.php]: should check if fully resolved here

**Comment by Nyholm** [comment]
Hm.. I dont think I like this. Im not sure you should care if the full response is downloaded. I think we could make some changes to #417 to take care of this problem.

**Comment by Nyholm** [comment]
I might be wrong about this...

**Comment by Nyholm** [comment]
I think this should be closed. It makes sense to have it when downloading more than one response. 
https://github.com/async-aws/aws/pull/417#discussion_r401197603

If you really really really want to use this. Then you can do `Result::multiplex([$result], 2.0);`

**Comment by jderusse** [comment]
closed in favor of #417

**Pull request closed without merging**